components/formats-gpl/src/loci/formats/in/ZeissCZIReader.java

```diff
@@ -1393,6 +1397,10 @@ private void readSegments(String id) throws IOException {
     if (in != null) {
       in.close();
     }
+    File f = new File(id);
+    if (!f.exists()){
+    		id = id.substring(0, id.lastIndexOf('.'));
+    }
```

Comment:
If both ```file``` and ```file.czi``` exist, though, this will still read from ```file.czi``` independent of which file was passed to setId.  For example:

```
$ cp data_repo/zeiss-czi/aaron/101post.czi ~/test
$ cp data_repo/zeiss-czi/jan/raw.czi ~/test.czi
$ showinf -nopix ~/test # reports 4 series
$ showinf -nopix data_repo/zeiss-czi/aaron/101post.czi # reports 1 series
```

It also requires the ```id``` to be a file on disk, which definitely is not always the case, see https://github.com/openmicroscopy/bioformats/pull/2916/files#diff-32810338543d500a122f15bccbcb1712R1447